server/job-workers/job-worker-server.cpp

```diff
@@ -129,10 +133,13 @@ int JobWorkerServer::job_parse_execute(connection *c) {
     assert(errno == EWOULDBLOCK);
     // another job worker has already taken the job (all job workers are readers for this fd)
     // or there are no more jobs in pipe
+    tvkprintf(job_workers, 3, "No jobs in pipe after wakeup\n");
     ++vk::singleton<SharedMemoryManager>::get().get_stats().job_worker_skip_job_due_steal;
+    rearm_read_job_fd(); // because > 1 workers can wake up on single job
```

Comment:
I think it worth to wrap rearm_read_job_fd() call in some RAII class. Currently it is possible forget to made such call and fd will hang.

Reply:
done